### ISSUE TITLE
Import users with Hash Algorithm PBKDF2_SHA256 and Rounds > 30000

### ISSUE THREAD

**Issue by dalenguyen** [firebase/firebase-tools#587]
I tried to import users with the hash PBKDF2_SHA256, and rounds > 100000. Then I got an error saying that: 

`Error: Must provide valid rounds(0..30000) for hash algorithm pbkdf2_sha256`

When I check the code in the _accountImporter.js_, firebase doesn't allow the numbers of rounds which is larger than 30000. 

```
case 'PBKDF2_SHA256':
    var roundsNum = parseInt(options.rounds, 10);
    if (isNaN(roundsNum) || roundsNum < 0 || roundsNum > 30000) {
      return utils.reject('Must provide valid rounds(0..30000) for hash algorithm ' + options.hashAlgo, {exit: 1});
    }
    return {hashAlgo: hashAlgo, rounds: options.rounds, valid: true};
```

Do you know how I can tackle this issue? 

Thanks,

**Comment by merlinnot** [comment]
It might be a backend limitation (someone from Google must confirm).

If so, you will need to somehow overcome this limitation. The easiest way is to create an endpoint which takes a login and password, calculates the hash using your existing algorithm, compares it with the current value for this user and if it matches you can create a Firebase user account for him using the password he has just provided (see https://firebase.google.com/docs/reference/admin/node/admin.auth.Auth#createUser).

That's a safe and easy solution. You can even speed up the login process by returning a token on successful account creation, see https://firebase.google.com/docs/reference/admin/node/admin.auth.Auth#createCustomToken

Let it work for a month or so, depending on how often your users typically sign in to the app, then you can get rid of this temporary solution and enforce the rest of the users to reset their passwords.

**Comment by dalenguyen** [comment]
Thanks @merlinnot, I will try other solutions then. I also sent a ticket to Google. Let's see if they can do anything about it.

**Comment by mbleigh** [comment]
This is best resolved via the ticket you sent to support -- the CLI just enforces backend requirements here.

**Comment by kmcnellis** [comment]
This is now fixed as of #664
